plugins/modules/inventory_intent.py

```diff
@@ -1151,6 +1216,199 @@ def provisioned_wired_device(self):
+            It iterates through each device, retrieves provisioning parameters using the get_wireless_param function,
+            and then calls the Cisco DNA Center API for wireless provisioning. If all devices are already provisioned,
+            it returns success with a relevant message.
+        """
+
+        provision_count, already_provision_count = 0, 0
+
+        for device_ip in device_ips:
+            try:
+                # Collect the device parameters from the playbook to perform wireless provisioing
+                provisioning_params = self.get_wireless_param(device_ip)
+
+                # Now we have provisioning_param so we can do wireless provisioning
+                response = self.dnac_apply['exec'](
+                    family="wireless",
+                    function="provision",
+                    op_modifies=True,
+                    params=provisioning_params,
+                )
+
+                if response.get("status") == "failed":
+                    description = response.get("description")
+                    error_msg = "Cannot do Provisioning for Wireless device {0} beacuse of {1}".format(device_ip, description)
+                    self.log(error_msg)
+                    continue
+
+                task_id = response.get("taskId")
+
+                while True:
+                    execution_details = self.get_task_details(task_id)
+                    progress = execution_details.get("progress")
+                    self.log(progress)
+
+                    if 'TASK_PROVISION' in progress:
+                        self.status = "success"
+                        self.result['changed'] = True
+                        self.result['response'] = execution_details
+                        provision_count += 1
+                        break
+                    elif execution_details.get("isError"):
+                        self.status = "failed"
+                        failure_reason = execution_details.get("failureReason")
+                        if failure_reason:
+                            self.msg = "Wireless Device Provisioning get failed because of {0}".format(failure_reason)
+                        else:
+                            self.msg = "Wireless Device Provisioning get failed"
+                        break
+            except Exception as e:
+                # Not returning from here as there might be possiblity that for some devices it comes into exception
+                # but for others it gets provision successfully or If some devices are already provsioned
+                error_message = "Error while Provisioning the device {0} in Cisco DNA Center - {1}".format(device_ip, str(e))
+                self.log(error_message)
+
+                if "already provisioned" in str(e):
+                    self.log(str(e))
+                    already_provision_count += 1
+
+        # Check If all the devices are already provsioned, return from here only
```

Comment:
```
   if already_provision_count == len(device_ips):
        self.handle_all_already_provisioned(device_ips)
    elif provision_count == len(device_ips):
        self.handle_all_provisioned()
    elif provision_count == 0:
        self.handle_all_failed_provision()
    else:
        self.handle_partially_provisioned(provision_count)

```


And APIs

```
def handle_all_already_provisioned(self, device_ips):
    self.status = "success"
    self.msg = "All the Wireless Devices given in the playbook are already Provisioned"
    self.log(self.msg)
    self.result['response'] = self.msg
    self.result['changed'] = False

def handle_all_provisioned(self):
    self.status = "success"
    self.log("All Wireless Devices provisioned successfully!!")

def handle_all_failed_provision(self):
    self.status = "failed"
    self.log("Wireless Device Provisioning failed for all devices")

def handle_partially_provisioned(self, provision_count):
    self.status = "success"
    self.result['changed'] = True
    self.log(f"Wireless Devices provisioned successfully partially for {provision_count} devices")

```

So we can have all the handling functions seperately and the code looks more readable

Reply:
(or), you can have one handling function and you can pass an argument saying
successful_provisioning
failed_provisioning
all_already_provisioned
all_provisioned
Something like that and based on the argument we can set status, result and logs..

Reply:
added new api and changed in both provisioned_wired_device and provisioned_wireless_device API's.